### PR TITLE
feat(TDI-45778): update mongodb cosmosdb tck.

### PR DESCRIPTION
**Pull request by vyu-talend**
feat(TDI-45778): upgrade mongo driver for aws documentdb.

### CODE REVIEW

File: mongodb/pom.xml [hunk 1]
```diff
@@ -19,6 +19,8 @@
     </modules>
 
     <properties>
-        <mongodb.client.version>3.12.11</mongodb.client.version>
+        <mongodb-driver-sync.version>4.10.2</mongodb-driver-sync.version>
+        <mongodb-driver-core.version>4.10.2</mongodb-driver-core.version>
```
Comment (jzhao-talend) [inline mongodb/pom.xml:23]: why can't use same version property?

File: mongodb/mongodb-common/src/main/java/org/talend/components/mongo/service/MongoCommonService.java [hunk 2]
```diff
@@ -53,81 +54,69 @@ public class MongoCommonService {
...
-                 * getOptions(datastore));
-                 * }
-                 */
-            }
-            return null;
+            client = MongoClients.create(getMongoClientSettings(datastore));
         } catch (Exception e) {
             // TODO use i18n
             LOG.error(e.getMessage(), e);
-            throw new RuntimeException(e);
```
Comment (jzhao-talend) [inline mongodb/mongodb-common/src/main/java/org/talend/components/mongo/service/MongoCommonService.java:91]: why remove throw exception?

**Review by jzhao-talend — APPROVED** [review]
LGTM

**Review by jzhao-talend — APPROVED** [review]
LGTM, but please take care the test coverage in [TDI-50743](https://jira.talendforge.org/browse/TDI-50743)

**Comment by vyu-talend** [comment]
> LGTM, but please take care the test coverage in [TDI-50743](https://jira.talendforge.org/browse/TDI-50743)

OK, I will improve it.

**Pull request merged**